planex/cmd/fetch.py

```diff
@@ -213,20 +237,11 @@ def fetch_source(args):
     except KeyError as exn:
         sys.exit("%s: No source corresponding to %s" % (sys.argv[0], exn))
 
-    url = urlparse.urlparse(resource.url)
-    if url.scheme in SUPPORTED_URL_SCHEMES:
-        fetch_url(url, resource.path, args.retries + 1)
-
-    elif url.scheme == 'ssh':
-        fetch_repo(url, resource)
-
-    elif url.scheme in ['', 'file'] and url.netloc == '':
-        shutil.copyfile(url.path, resource.path)
-        write_originfile(resource.path, url.path)
-
-    else:
+    try:
+        fetch_source_dispatch(resource, args.retries)
+    except UnsupportedScheme as exn:
         sys.exit("%s: Unsupported url scheme %s" %
-                 (sys.argv[0], url.scheme))
+                 (sys.argv[0], exn.message))
```

Comment:
`exception.message` is deprecated

https://www.python.org/dev/peps/pep-0352/#retracted-ideas